test/e2e/alertmanager_e2e_test.go

```diff
@@ -79,3 +83,111 @@ func TestAlertmanagerVersionMigration(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestExposingAlertmanagerWithNodePort(t *testing.T) {
+	alertmanager := framework.MakeBasicAlertmanager("test-alertmanager", 1)
+	alertmanagerService := framework.MakeAlertmanagerNodePortService(alertmanager.Name, "nodeport-service")
+
+	defer func() {
+		if err := framework.DeleteAlertmanagerAndWaitUntilGone(alertmanager.Name); err != nil {
+			t.Fatal(err)
+		}
+		if err := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).Delete(alertmanagerService.Name, nil); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	if err := framework.CreateAlertmanagerAndWaitUntilReady(alertmanager); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).Create(alertmanagerService); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := framework.WaitForServiceReady(time.Minute, alertmanagerService.Name); err != nil {
```

Comment:
maybe we can merge these with the above function to keep the theme of `CreateXAndWaitUntilReady`